component/src/main/java/org/wso2/extension/siddhi/execution/rdbms/CUDStreamProcessor.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.rdbms;
+
+import com.zaxxer.hikari.HikariDataSource;
+import org.wso2.extension.siddhi.execution.rdbms.util.RDBMSStreamProcessorUtil;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.SystemParameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.event.stream.populater.ComplexEventPopulater;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.query.processor.Processor;
+import org.wso2.siddhi.core.query.processor.stream.StreamProcessor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This extension can be used to perform SQL CUD (INSERT, UPDATE, DELETE) queries on a wso2 datasource.
+ */
+@Extension(
+        name = "cud",
+        namespace = "rdbms",
+        description = "The function can be used to perform SQL CUD (INSERT, UPDATE, DELETE) queries on a wso2 " +
+                "datasource. \nNote: This will only work within a carbon environment.\n",
+        parameters = {
+                @Parameter(
+                        name = "datasource.name",
+                        description = "The name of the wso2 datasource the query should be performed on",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "query",
+                        description = "The select query that needs to be performed",
```

Comment:
The select query __(formatted according to the appropriate database type)__ that needs to be performed

Reply:
Fixed in 84cce6e

Reply:
This should be update, delete, or insert query not select

Reply:
Fixed in 663e63b